Use Starlight `sidebar` user-config format for `<StarlightPage />` `sidebar` prop (#2168)
withastro/starlight: .changeset/chilled-kiwis-count.md

---
'@astrojs/starlight': minor
---

> ⚠️ **BREAKING CHANGE:** Updates the `<StarlightPage />` component `sidebar` prop to accept an array of [`SidebarItem`](https://starlight.astro.build/reference/configuration/#sidebaritem) like the Starlight global `sidebar` configuration.

Reviewer: I think we should probably pluralize “item” here to read a bit better, and also suggesting an alternative to just “global” to clarify things:

```suggestion
⚠️ **BREAKING CHANGE:** Updates the `<StarlightPage />` component `sidebar` prop to accept an array of [`SidebarItem`](https://starlight.astro.build/reference/configuration/#sidebaritem)s like the main Starlight `sidebar` configuration in `astro.config.mjs`.
```

This change simplifies the definition of sidebar items in the `<StarlightPage />` component, allows for shared sidebar configuration between the global `sidebar` option and `<StarlightPage />` component, and also enables the usage of autogenerated sidebar groups with the `<StarlightPage />` component.
If you are using the `<StarlightPage />` component with a custom `sidebar` configuration, you will need to update the `sidebar` prop to an array of [`SidebarItem`](https://starlight.astro.build/reference/configuration/#sidebaritem) objects.

For example, the following custom page with a custom `sidebar` configuration defines a “Resources” group with a “New” badge, a link to the “Showcase” page which is part of the `docs` content collection, and a link to the Starlight website:

> ```tsx

Reviewer: Shouldn’t these use Astro?

````suggestion
```astro
````

---
// src/pages/custom-page/example.astro
---

<StarlightPage
	frontmatter={{ title: 'My custom page' }}
	sidebar={[
		{
			type: 'group',
			label: 'Resources',
			badge: { text: 'New' },
			items: [
				{ type: 'link', label: 'Showcase', href: '/showcase/' },
				{ type: 'link', label: 'Starlight', href: 'https://starlight.astro.build/' },
			],
		},
	]}
>
	<p>This is a custom page with a custom component.</p>
</StarlightPage>
```

This configuration will now need to be updated to the following:

> ```tsx

Reviewer: ````suggestion
```astro
````
